### PR TITLE
Extend sign up form

### PR DESCRIPTION
**Pull request by mikemike27**
Extend sign up form by including user first name, last name and date of birth(date, month, year). And then, register the info into cognito.

### CODE REVIEW

File: src/app/components/formElements/Dropdown/dropdown.module.css [hunk 1]
```diff
@@ -62,6 +62,7 @@
   font-family: inherit;
   outline: none;
   padding: 8px 15px;
+  width: 10px;
```
[the remark this answers is not in the document]
Comment (mikemike27) [reply src/app/components/formElements/Dropdown/dropdown.module.css:65]: without the width, the clearbutton and dropdown svg will go outside the input

[the remark this answers is not in the document]
Comment (salman0719) [reply src/app/components/formElements/Dropdown/dropdown.module.css:65]: Try to see if `width: 100%` has any issues. If not, set it to 100%.

File: src/app/utilities/auth/constants.ts [hunk 2]
```diff
@@ -48,3 +48,28 @@ export const DEFAULT_USER = {
...
 export const USER_TABLE_PARTITION_ID = 'Email';
 export const DEFAULT_SIGN_IN_REDIRECT_URL = '/profile';
+
+export const OPTIONS_DATE = Array.from({ length: 31 }, (_, i) => ({
+  label: (i + 1).toString(),
+  value: i + 1,
+}));
+
+export const OPTIONS_MONTH = [
+  { label: 'Jan', value: '1' },
```
Comment (salman0719) [inline src/app/utilities/auth/constants.ts:58]: Use `number` type for `OPTIONS_MONTH`'s `value`s. `Date()` constructor probably expects `monthIndex`, so the value should start from '0`.

File: src/app/components/auth/AuthInitSignUp.tsx [hunk 3]
```diff
@@ -95,8 +115,22 @@ const AuthInitSignUp: React.FC = () => {
       <Form
         methods={methods}
         onSubmit={methods.handleSubmit(onSubmit)}
-        className={classNames(isConfirming && 'hide')}
+        className={classNames('row', { hide: isConfirming })}
```
Comment (salman0719) [inline src/app/components/auth/AuthInitSignUp.tsx:118]: `<Form />` probably is already rendering the `row` class by default. Check if it does, and if so, remove the `row` from here.

File: src/app/components/auth/AuthInitSignUp.tsx [hunk 4]
```diff
@@ -131,6 +165,33 @@ const AuthInitSignUp: React.FC = () => {
...
+          cols={FORM_ELEMENT_COL_WIDTH.SMALL}
+          options={OPTIONS_MONTH}
+        />
+        <Dropdown
+          name='year'
+          label='Year'
+          placeholder='Year'
+          cols={FORM_ELEMENT_COL_WIDTH.SMALL}
+          options={OPTIONS_YEAR}
+        />
```
Comment (salman0719) [inline src/app/components/auth/AuthInitSignUp.tsx:194]: Use `radioMode` (so that users can select only one) for all the dropdowns.

**Review by salman0719 — CHANGES_REQUESTED** [review]

**Pull request merged**